client/components/tinymce/plugins/mentions/mentions.jsx

```diff
@@ -0,0 +1,330 @@
+		const { editor, node } = this.props;
+		const mceToolbar = tinymce.$( '.mce-toolbar-grp', editor.getContainer() )[ 0 ];
+		const range = editor.selection.getRng();
+		const rectList = range.getClientRects();
+		let height;
+
+		if ( rectList.length > 0 ) {
+			height = rectList[ 0 ].height;
+		} else {
+			height = editor.selection.getNode().offsetHeight;
+		}
+
+		this.left = left;
+		this.top = top;
+
+		node.style.left = `${ this.left }px`;
+		// 10 is the top position of .mentions__suggestions .popover__inner, which hasn't rendered yet.
+		node.style.top = `${ mceToolbar.offsetHeight + this.top + height - 10 }px`;
+	}
+
+	getQueryText() {
+		// (?:^|\\s) means start of input or whitespace
+		// ([A-Za-z0-9_\+\-]*) means 0 or more instances of: A-Z a-z 0-9 _ + -
+		const matcher = new RegExp( '(?:^|\\s)@([A-Za-z0-9_\+\-]*)$', 'gi' );
+		const range = this.props.editor.selection.getRng();
+		const textBeforeCaret = range.startContainer.textContent.slice( 0, range.startOffset );
+		const match = matcher.exec( textBeforeCaret );
+
+		if ( match && ( match.length > 1 ) ) {
+			return match[ 1 ];
+		}
+
+		return null;
+	}
+
+	getSelectedSuggestionIndex() {
+		if ( ! this.matchingSuggestions ) {
+			return null;
+		}
+
+		if ( ! this.state.selectedSuggestionId ) {
+			return 0;
+		}
+
+		for ( let i = 0; i < this.matchingSuggestions.length; i++ ) {
+			if ( this.matchingSuggestions[ i ].ID === this.state.selectedSuggestionId ) {
+				return i;
+			}
+		}
+
+		return null;
+	}
+
+	getSuggestion = function() {
+		if ( ! this.matchingSuggestions ) {
+			return null;
+		}
+
+		if ( ! this.state.selectedSuggestionId && this.matchingSuggestions.length > 0 ) {
+			return this.matchingSuggestions[ 0 ];
+		}
+
+		for ( let i = 0; i < this.matchingSuggestions.length; i++ ) {
+			if ( this.matchingSuggestions[ i ].ID === this.state.selectedSuggestionId ) {
+				return this.matchingSuggestions[ i ];
+			}
+		}
+
+		return null;
+	}
+
+	insertSuggestion = ( suggestion ) => {
+		const { editor } = this.props;
+		const re = /@\S*/;
+		const markup = <EditorMention username={ suggestion.user_login } />;
+		const range = editor.selection.getRng();
+		const mentionRange = document.createRange();
+
+		// Set a range for the user-entered mention.
+		mentionRange.setStart( range.startContainer, Math.max( range.startOffset - ( this.state.query.length + 1 ), 0 ) );
+		mentionRange.setEnd( range.endContainer, range.endOffset );
+		editor.selection.setRng( mentionRange );
+
+		// Replace user-entered mention with full username.
+		editor.selection.setContent( editor.selection.getContent().replace( re, ReactDomServer.renderToStaticMarkup( markup ) ) );
+
+		editor.getBody().focus();
+	}
+
+	onKeyDown = ( e ) => {
+		const { keyCode } = e;
+
+		if ( includes( [ VK.DOWN, VK.UP ], keyCode ) ) {
+			const selectedIndex = this.getSelectedSuggestionIndex();
+			let nextIndex;
+
+			if ( null === selectedIndex ) {
+				return;
+			}
+
+			if ( keyCode === VK.DOWN ) {
```

Comment:
Maybe move the prevent default to above this line? 

Also a switch might read a little nicer here, though that's optional.